src/state/opamStateConfig.mli

```diff
@@ -137,3 +137,9 @@ module Repos : sig
   val safe_read:
     ?lock_kind: 'a lock -> 'b global_state -> OpamFile.Repos_config.t
 end
+
+(* Raw read an switch config to downgrade its [opam-version] from 2.1 to 2.0.
+   It is necessary to handle opam root and switch upgrade from 2.1
+   intermediates roots to 2.1. *)
```

Comment:
I'd add more context here to explain _why_ it is necessary ? E.g. "This allows a workaround for a bug in versions 2.1~xx which wrongly updated the declared switch versions, requiring that we fix it during [OpamFormatUpdate] from these specific intermediate versions"